frame/frame-layer.go

```diff
@@ -60,6 +61,20 @@ func NewFrameLayer(transportLayer io.ReadWriter) *Layer {
 	go frameLayer.bgWork()
 	go frameLayer.bgRead()
 
+	// Empty the buffer
+empty:
+	for {
+		select {
+		case msg := <-parserInput:
```

Comment:
Instead of relying on frameLayer.bgRead, could we perhaps consume what is on transportLayer instead?